kubernetes/spack/internal-spack-io/gh-gl-sync/config-maps.yaml

```diff
@@ -0,0 +1,218 @@
+    parser.add_argument("-t", "--token", help="GitHub access token")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-o", "--open", help="List open PRs",
+                       action="store_true", default=False)
+    group.add_argument("-c", "--closed", help="List closed PRs",
+                       action="store_true", default=False)
+    group.add_argument("-a", "--all", help="List all PRs",
+                       action="store_true", default=False)
+    parser.add_argument("repo", help="GitHub repo (org/repo or user/repo)")
+    args = parser.parse_args()
+
+    if args.open:
+        state = "open"
+    elif args.closed:
+        state = "closed"
+    elif args.all:
+        state = "all"
+    else:
+        state = "open"
+
+    try:
+        request = urllib.request.Request(
+            'https://api.github.com/repos/%s/pulls?state=%s' % (args.repo, state))
+        if args.token:
+            request.add_header('Authorization', 'token %s' % args.token)
+        response = urllib.request.urlopen(request)
+    except OSError:
+        sys.exit(1)
+
+    prs = json.loads(response.read())
+
+    for pr in prs:
+        print("%d %s" % (pr['number'], pr['head']['ref']))
+
+  github-prs-to-gitlab.sh: |
+    #!/usr/bin/env bash
+
+    set -e
+
+    CLEANUP_SSH_AGENT=0
+    CLEANUP_WORKDIR=0
+
+    function cleanup()
+    {
+      echo "Cleanup:"
+      if [ ${CLEANUP_WORKDIR} -eq 1 ]
+      then
+        echo "  Removing ${TMP_WORKDIR}"
+        rm -rf "${TMP_WORKDIR}"
+      fi
+      if [ ${CLEANUP_SSH_AGENT} -eq 1 ]
+      then
+        echo "  Shutting down ssh-agent(${SSH_AGENT_PID})"
+        eval $(ssh-agent -k)
+      fi
+    }
+
+
+    if [ $# -ne 3 ]
+    then
+      echo "Usage: $0 <github-repo> <gitlab-host> <gitlab-repo>"
+      exit 1
+    fi
+
+    GITHUB_REPO=$1
+    GITLAB_HOST=$2
+    GITLAB_REPO=$3
+    GITLAB_URL="git@${GITLAB_HOST}:${GITLAB_REPO}.git"
+    BASEDIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+    if [ -z "${GITLAB_SSH_KEY_BASE64}" ]
+    then
+      echo "Error: GITLAB_SSH_KEY_BASE64 is empty"
+      exit 1
+    fi
+
+    trap cleanup EXIT
+
+    # Start the ssh agent
+    eval $(ssh-agent -s)
+    CLEANUP_SSH_AGENT=1
+
+    # Add the necessary key
+    echo "${GITLAB_SSH_KEY_BASE64}" | base64 -d | tr -d '\r' | ssh-add -
+
+    git --version
+
+    export GIT_SSH_COMMAND="ssh -F /dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
+
+    # Setup the local repo with two remotes
+    TMP_WORKDIR=$(mktemp -d)
+    CLEANUP_WORKDIR=1
+    echo "Initializing local repo in ${TMP_WORKDIR}"
+    cd ${TMP_WORKDIR}
+    git init
+    git remote add github https://github.com/${GITHUB_REPO}.git
+    git remote add gitlab ${GITLAB_URL}
+
+    # Retrieve open PRs
+    OPEN_PR_BRANCHES="$(python3 ${BASEDIR}/github-list-prs.py -o ${GITHUB_REPO} --token ${GITHUB_TOKEN} | awk '{print "pr"$1"_"$2}' | sort -r)"
```

Comment:
IIRC the repo is a positional argument and needs to be specified last, i.e. `github-list-prs.py -t ${GITHUB_TOKEN} -o ${GITHUB_REPO}`.  The use of the token is also entirely optional.  It's helpful to get around API rate limits from a given IP but since this is being run in a cloud container I don't this that's really a concern is it?